common/static/js/src/tooltip_manager.js

```diff
@@ -1,109 +1,239 @@
 (function() {
     'use strict';
-    var TooltipManager = function (element) {
-        this.element = $(element);
-        // If tooltip container already exist, use it.
-        this.tooltip = $('div.' + this.className.split(/\s+/).join('.'));
-        // Otherwise, create new one.
-        if (!this.tooltip.length) {
-            this.tooltip = $('<div />', {
-                'class': this.className
-            }).appendTo(this.element);
-        }
-
-        this.hide();
-        _.bindAll(this, 'show', 'hide', 'showTooltip', 'moveTooltip', 'hideTooltip', 'click');
-        this.bindEvents();
-    };
 
-    TooltipManager.prototype = {
-        // Space separated list of class names for the tooltip container.
+    var Tooltip = Backbone.View.extend({
         className: 'tooltip',
-        SELECTOR: '[data-tooltip]',
 
-        bindEvents: function () {
-            this.element.on({
-                'mouseover.TooltipManager': this.showTooltip,
-                'mousemove.TooltipManager': this.moveTooltip,
-                'mouseout.TooltipManager': this.hideTooltip,
-                'click.TooltipManager': this.click
-            }, this.SELECTOR);
+        template: _.template('<div class="tooltip-content"><%= content %></div><div class="tooltip-arrow">▾</div>'),
+
+        events: {
+            'mouseenter': 'show',
+            'mouseleave': 'hide',
+            'focusout': 'focusout'
         },
 
-        getCoords: function (pageX, pageY) {
-            return {
-                'left': pageX - 0.5 * this.tooltip.outerWidth(),
-                'top': pageY - (this.tooltip.outerHeight() + 15)
-            };
+        initialize: function(content) {
+            this.content = content;
+            this.timer = null;
         },
 
-        show: function () {
-            this.tooltip.show().css('opacity', 1);
+        // Display this tooltip.
+        show: function() {
+            if (this.timer) {
+                clearTimeout(this.timer);
+            }
+            this.timer = null;
+            this.$el.show().css('opacity', 1);
         },
 
-        hide: function () {
-            this.tooltip.hide().css('opacity', 0);
+        // Hide this tooltip. Includes a delay before the tooltip starts fading
+        // out, and a second delay to allow the tooltip to fade out. Once the
+        // tooltip is hidden, it is removed from the DOM, and the `removed`
+        // event is triggered. If `show` is called before the tooltip is
+        // removed, it is redisplayed.
+        hide: function() {
+            if (this.timer) {
+                clearTimeout(this.timer);
+            }
+            var remove = _.bind(this.remove, this);
```

Comment:
Note (no change needed): edX doesn't support older browsers so it's fine to use native javascript `bind()` if you want, rather than the underscore version.